test/model/content_test.dart

```diff
@@ -125,6 +125,78 @@ class ContentExample {
     const ImageEmojiNode(
       src: '/static/generated/emoji/images/emoji/unicode/zulip.png', alt: ':zulip:'));
 
+  static const spoilerDefaultHeader = ContentExample(
+    'spoiler with default header',
+    '```spoiler\nhello world\n```',
+    expectedText: 'Spoiler', // or a translation
+    '<div class="spoiler-block"><div class="spoiler-header">\n'
+        '</div><div class="spoiler-content" aria-hidden="true">\n'
+        '<p>hello world</p>\n'
+        '</div></div>',
+    [SpoilerNode(
+      header: [],
+      content: [ParagraphNode(links: null, nodes: [TextNode('hello world')])],
+    )]);
+
+  static const spoilerPlainCustomHeader = ContentExample(
+    'spoiler with plain custom header',
+    '```spoiler hello\nworld\n```',
+    expectedText: 'hello',
+    '<div class="spoiler-block"><div class="spoiler-header">\n'
+        '<p>hello</p>\n'
+        '</div><div class="spoiler-content" aria-hidden="true">\n'
+        '<p>world</p>\n'
+        '</div></div>',
+    [SpoilerNode(
+      header: [ParagraphNode(links: null, nodes: [TextNode('hello')])],
+      content: [ParagraphNode(links: null, nodes: [TextNode('world')])],
+    )]);
+
+  static const spoilerRichHeaderAndContent = ContentExample(
+    'spoiler with rich header and content',
+    '```spoiler 1. * ## hello\n*italic* [zulip](https://zulip.com/)\n```',
+    expectedText: 'hello',
+    '<div class="spoiler-block"><div class="spoiler-header">\n'
+        '<ol>\n<li>\n<ul>\n<li>\n<h2>hello</h2>\n</li>\n</ul>\n</li>\n</ol>\n</div>'
+        '<div class="spoiler-content" aria-hidden="true">\n'
+        '<p><em>italic</em> <a href="https://zulip.com/">zulip</a></p>\n'
+        '</div></div>',
+    [SpoilerNode(
+      header: [ListNode(ListStyle.ordered, [
+          [ListNode(ListStyle.unordered, [
+            [HeadingNode(level: HeadingLevel.h2, links: null, nodes: [
+              TextNode('hello'),
+            ])]
+          ])],
+        ])],
```

Comment:
nit: two-space indents, here and below
```suggestion
      header: [ListNode(ListStyle.ordered, [
        [ListNode(ListStyle.unordered, [
          [HeadingNode(level: HeadingLevel.h2, links: null, nodes: [
            TextNode('hello'),
          ])]
        ])],
      ])],
```